File: apps/backend-node/src/features/payments/mappers/payment.mapper.ts
import type { payments } from '@prisma/client';

import type { ConfirmPaymentResult, CreatePaymentOrderResult } from '../dto';
import { PaymentStatus } from '../dto';

export function mapCreatePaymentOrderResult(input: {
  paymentId: string;
  orderId: string;
  bookingId: string;
  amountKrw: number;
  orderName: string;
}): CreatePaymentOrderResult {
  return {
    paymentId: input.paymentId,
    orderId: input.orderId,
    bookingId: input.bookingId,
    amountKrw: input.amountKrw,
    orderName: input.orderName,
  };
}

export function mapConfirmPaymentResult(
  payment: payments,
  bookingId?: string | null,
): ConfirmPaymentResult {
  return {
    paymentId: payment.id,
    orderId: payment.order_id,
    bookingId: bookingId !== undefined ? bookingId : payment.booking_id,
    status: payment.status as PaymentStatus,
  };
}

File: apps/backend-java/src/main/java/com/housingplatform/features/payments/mapper/PaymentMapper.java
package com.housingplatform.features.payments.mapper;

import com.housingplatform.features.payments.dto.ConfirmPaymentResult;
import com.housingplatform.features.payments.dto.CreatePaymentOrderResult;
import com.housingplatform.persistence.entity.Payment;
import java.util.UUID;

public final class PaymentMapper {

  private PaymentMapper() {}

  public static CreatePaymentOrderResult toCreateOrderResult(
      UUID paymentId, UUID orderId, UUID bookingId, int amountKrw, String orderName) {
    return new CreatePaymentOrderResult(paymentId, orderId, bookingId, amountKrw, orderName);
  }

  public static ConfirmPaymentResult toConfirmResult(Payment payment) {
    return new ConfirmPaymentResult(
        payment.getId(), payment.getOrderId(), payment.getBookingId(), payment.getStatus());
  }

  public static ConfirmPaymentResult toConfirmResult(
      UUID paymentId, UUID orderId, UUID bookingId, com.housingplatform.persistence.enums.PaymentStatus status) {
    return new ConfirmPaymentResult(paymentId, orderId, bookingId, status);
  }
}
